src/hooks/useSelectInterest.js

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+
+export default function useSelectInterest(api) {
+  const [availableTopics, setAvailableTopics] = useState(null);
+  const [subscribedTopics, setSubscribedTopics] = useState(null);
+  const [allTopics, setAllTopics] = useState(null);
+  const [subscribedSearches, setSubscribedSearches] = useState(null);
+  const [showingSpecialInterestModal, setshowingSpecialInterestModal] =
+    useState(false);
+
+  useEffect(() => {
+    api.getAvailableTopics((data) => {
+      setAvailableTopics(data);
+    });
+
+    api.getSubscribedTopics((data) => {
+      setSubscribedTopics(data);
+    });
+
+    //custom interest filters
+    api.getSubscribedSearchers((data) => {
+      setSubscribedSearches(data);
+    });
+  }, [api]);
+
+  useEffect(() => {
+    if (availableTopics && subscribedTopics) {
+      let newAllTopics = [...availableTopics, ...subscribedTopics];
+      newAllTopics.sort((a, b) => a.title.localeCompare(b.title));
+      setAllTopics(newAllTopics);
+    }
+  }, [availableTopics, subscribedTopics]);
+
+  if (
+    !availableTopics ||
+    !subscribedTopics ||
+    !subscribedSearches ||
+    !allTopics
+  )
+    return "";
```

Comment:
this pattern is a bit strange. given that we're expecting the hook to be used with a deconstructor operator, what are the various variables going to unpack to when we just return an empty string?

Reply:
Very good observation - I have initially left this part of the previous interest selection implementation, but I now agree that this condition looked confusing. It is outside the hooks, and I believe it was meant to handle the initial states of all collections (`availableTopics`, `subscribedTopics`, `allTopics`, and `subscribedSearches`). The `map` function and **deconstructor** would throw errors when the initial state was `null` or `undefined`, so this was probably a workaround to prevent errors before the data was fetched from the backend. 

<br>I have changed the code, so we no longer need this state adjustment. The **initial state** of all collections in the `useSelectInterest` hook is now an empty array `[]`. This prevents runtime errors and ensures that the collections are compatible with the `map` function and **deconstructor** from the very beginning, even before the data is fetched.